Page teka

tRy-HaRd-tgn opened the pull request.

n3v3sl33p committed:
add backgound container, start nav-bar

n3v3sl33p committed:
nav-bar

tRy-HaRd-tgn committed:
feat:Создал страницу

tRy-HaRd-tgn committed:
feat: сделал input с рандомными placeholder

tRy-HaRd-tgn committed:
feat: установил библиотеку, попытался сделать
виджет

tRy-HaRd-tgn committed:
feat: сделал виджет
feat: сделал страницу

Hubulbib commented on src/shared/ui/teka-input/index.js:
```diff
@@ -0,0 +1,20 @@
+import { useEffect } from "react";
```
Методы из одной библиотеки можно скомпоновать, просто перечислив их.

Hubulbib commented on src/shared/ui/teka-input/index.js:
```diff
@@ -0,0 +1,20 @@
+import { useState } from "react";
+export const TekaInput = ({ img, ...props }) => {
+  const [index, setIndex] = useState();
+  const array = ["Что хочешь попробовать", "Какую кофейню ищите"];
```
Вынести эту массив-константу в сегмент **_model_** в файл search-placeholder-data.

Hubulbib commented on src/widgets/teka-widget/index.js:
```diff
@@ -0,0 +1,41 @@
+import styles from "./style.module.scss";
+import "slick-carousel/slick/slick.css";
+import "slick-carousel/slick/slick-theme.css";
+export const TekaWidget = ({ h, ...props }) => {
```
замени название props-а **h** на **label**

Hubulbib commented on src/widgets/teka-widget/index.js:
сделай разделение на ui, model. Текущий код из index.js перенеси в ui.

Hubulbib commented on src/widgets/teka-widget/index.js:
```diff
@@ -0,0 +1,41 @@
+import "slick-carousel/slick/slick.css";
+import "slick-carousel/slick/slick-theme.css";
+export const TekaWidget = ({ h, ...props }) => {
+  const settings = {
```
вынеси в сегмент model

Hubulbib commented on src/widgets/teka-widget/index.js:
```diff
@@ -0,0 +1,41 @@
+    <div className={styles.sliderContainer}>
+      <h2 className={styles.h}>{h}</h2>
+      <Slider {...settings}>
+        <div className={styles.div}>
```
сделай каждый этот `div `(элемент карусели) в отдельный компонент _shared_ **PostItem**.

Hubulbib commented on src/widgets/teka-widget/index.js:
```diff
@@ -0,0 +1,41 @@
+export const TekaWidget = ({ h, ...props }) => {
+  const settings = {
+    dots: false,
+    infinite: true,
```
сделать его конечным

tRy-HaRd-tgn committed:
fix:пофиксил импортирование методов из одной
библиотеки в несколько строчек
src/shared/ui/teka-input/index.js
fix: перенёс массив в отдельный сегмент model
в фвйл search-placeholder-data
fix: заменил название h на label
src/widgets/teka-widget/index.js
feat: сделал разделение на ui,model
src/widgets/teka-widget/index.js
fix: вынес настройки компонента библиотеки slider
в отдельный сегмент model
feat: добавил компонент PostItem
src/widgets/teak-widget/index.js
fix: сделал slider конечным
src/widgets/teka-widget/index.js

Hubulbib commented on src/pages/teka/index.js:
```diff
@@ -0,0 +1,17 @@
+import { TekaInput } from "../../shared/ui";
+import styles from "./style.module.scss";
+import { Search } from "../../shared/assets";
+import { TekaWidget } from "../../widgets/teka-widget/ui";
```
импортировать можно только через OpenAPI файл

Hubulbib commented on src/pages/profile/index.js:
```diff
@@ -0,0 +1,14 @@
+import React from "react";
```
использовать деструктуризацию:
`import {useState} from "react"`

tRy-HaRd-tgn replied:
это страничка темы

Hubulbib reviewed (changes requested):
Косметические правки

tRy-HaRd-tgn committed:
fix: изменил инпуты

Hubulbib reviewed (approved):
Отличная работа
